includes/class-wc-gateway-swedbank-pay-swish.php

```diff
@@ -346,120 +291,135 @@ public function process_payment( $order_id ) {
 		switch ( $this->method ) {
 			case 'redirect':
 				// Get Redirect
-				$redirect = self::get_operation( $result['operations'], 'redirect-sale' );
 
-				return [
+				return array(
 					'result'   => 'success',
-					'redirect' => $redirect
-				];
+					'redirect' => $result->getOperationByRel( 'redirect-sale' ),
+				);
 				break;
 			case 'direct':
 				// Sale payment
-				$sale = self::get_operation( $result['operations'], 'create-sale' );
-
 				try {
-					$params = [
-						'transaction' => [
-							'msisdn' => apply_filters( 'swedbank_pay_swish_phone_format', $phone, $order )
-						]
-					];
-
-					$result = $this->request( 'POST', $sale, $params );
+					$this->core->initiateSwishPaymentDirect(
+						$result->getOperationByRel( 'create-sale' ),
+						apply_filters( 'swedbank_pay_swish_phone_format', $order->get_billing_phone(), $order )
+					);
 				} catch ( \Exception $e ) {
-					$this->log( sprintf( '[ERROR] Create Sale: %s', $e->getMessage() ) );
 					wc_add_notice( $e->getMessage(), 'error' );
 
 					return false;
 				}
 
-				return [
+				return array(
 					'result'   => 'success',
-					'redirect' => $this->get_return_url( $order )
-				];
+					'redirect' => $this->get_return_url( $order ),
+				);
 
 				break;
 
 			default:
-				wc_add_notice( __( 'Wrong method', WC_Swedbank_Pay::TEXT_DOMAIN ), 'error' );
+				wc_add_notice( __( 'Wrong method', 'swedbank-pay-woocommerce-payments' ), 'error' );
 
 				return false;
 		}
+	}
+
+	/**
+	 * Process Refund
+	 *
+	 * If the gateway declares 'refunds' support, this will allow it to refund
+	 * a passed in amount.
+	 *
+	 * @param int $order_id
+	 * @param float $amount
+	 * @param string $reason
+	 *
+	 * @return  bool|wp_error True or false based on success, or a WP_Error object
+	 */
+	public function process_refund( $order_id, $amount = null, $reason = '' ) {
+		$order = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return false;
+		}
+
+		// Full Refund
+		if ( is_null( $amount ) ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function is_null() is discouraged; use strict comparison "=== null" instead.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)